app/src/main/java/org/jellyfin/mobile/bridge/NativeInterface.kt

```diff
@@ -118,21 +118,27 @@ class NativeInterface(private val context: Context) : KoinComponent {
     }
 
     @JavascriptInterface
-    fun downloadFile(args: String): Boolean {
-        val title: String
-        val filename: String
-        val url: String
-        try {
-            val options = JSONObject(args)
-            title = options.getString("title")
-            filename = options.getString("filename")
-            url = options.getString("url")
-        } catch (e: JSONException) {
-            Timber.e("Download failed: %s", e.message)
-            return false
+    fun downloadFiles(args: String): Boolean {
+        val files = JSONArray(args)
```

Comment:
This should be in the try/catch, instead. Probably for the best to cancel all downloads if one fails anyway.